### PR TITLE
MGMT-4091 - change test-infra to work with static network configurati…

### PR DESCRIPTION
**Pull request by yevgeny-shnaidman**
…on instead of static ips

### CODE REVIEW

File: discovery-infra/test_infra/tools/static_network.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+
+def generate_static_network_data_from_tf(tf_folder):
+    tfvars_json_file = os.path.join(tf_folder, consts.TFVARS_JSON_NAME)
+    with open(tfvars_json_file) as _file:
+        tfvars = json.load(_file)
+
+    network_config = []
+
+    for count in range(tfvars['master_count']):
```
Comment (masayag) [inline discovery-infra/test_infra/tools/static_network.py:22]: should there be another loop for worker_count ?

**Comment by empovit** [comment]
The dicts, often nested, make it difficult to understand the structure of data and work with it. @YuviGold isn't going to like it :)
What about static network config in day 2 flow?

File: discovery-infra/start_discovery.py [hunk 2]
```diff
@@ -474,7 +474,7 @@ def set_hosts_roles(client, cluster, nodes_details, machine_net, tf, master_coun
     if machine_net.has_ip_v4:
         libvirt_nodes = utils.get_libvirt_nodes_mac_role_ip_and_name(networks_names[0])
         libvirt_nodes.update(utils.get_libvirt_nodes_mac_role_ip_and_name(networks_names[1]))
-        if static_ip_mode:
+        if static_network_mode:
```
[the remark this answers is not in the document]
Comment (yevgeny-shnaidman) [reply discovery-infra/start_discovery.py:477]: fixed

File: discovery-infra/test_infra/tools/static_network.py (hunk 1 above)
Comment (yevgeny-shnaidman) [reply discovery-infra/test_infra/tools/static_network.py:22]: fixed

**Comment by yevgeny-shnaidman** [comment]
> The dicts, often nested, make it difficult to understand the structure of data and work with it. @YuviGold isn't going to like it :)
> What about static network config in day 2 flow?

will be added in MGMT-4360

File: discovery-infra/test_infra/tools/static_network.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        interface['ipv4'] = ipv4_dict
+    if ipv6_dict:
+        interface['ipv6'] = ipv6_dict
+
+    return interface
+
+
+def _prepare_ip_dict(cidr, host_count):
+    network = ip_network(cidr)
+    ip = str(ip_address(network.network_address) + 30 + host_count)
```
Comment (masayag) [inline discovery-infra/test_infra/tools/static_network.py:89]: pls consider storing the value '30' in a variable to describe its role (e.g. ip_address_offset)

**Review by masayag — COMMENTED** [review]
lgtm

**Pull request merged**